src/openms/include/OpenMS/ANALYSIS/ID/SiriusAdapterAlgorithm.h

```diff
@@ -109,68 +163,156 @@ namespace OpenMS
       @param feature_mapping: FeatureToMs2Indices with feature mapping
       @param spectra: Input of MSExperiment with spectra information
-      @param sirius_algo: Parameters needed for preprocessing
       */
-      static void checkFeatureSpectraNumber(const String& featureinfo,
-                                            const FeatureMapping::FeatureToMs2Indices& feature_mapping, 
-                                            const MSExperiment& spectra, 
-                                            const SiriusAdapterAlgorithm& sirius_algo);
+      void logFeatureSpectraNumber(const String& featureinfo,
+                                   const FeatureMapping::FeatureToMs2Indices& feature_mapping,
+                                   const MSExperiment& spectra);
 
-      
       /**
       @brief Call SIRIUS with QProcess
 
-      @return Vector with paths to a compound 
-
       @param tmp_ms_file: path to temporary .ms file
       @param tmp_out_dir: path to temporary output folder
       @param executable: path to executable
       @param out_csifingerid: path to CSI:FingerID output (can be empty).
-      @param sirius_algo: Parameters needed for SIRIUS
 
+      @return Vector with paths to a compound
       */
-      const static std::vector<String> callSiriusQProcess(const String& tmp_ms_file,
-                                                          const String& tmp_out_dir,
-                                                          String& executable,
-                                                          const String& out_csifingerid,
-                                                          const SiriusAdapterAlgorithm& sirius_algo);
-
-      // getter (used to call functions from SiriusMSConverter, SiriusMzTabWriter, CsiFingerIDMzTabWriter)
-      String getFeatureOnly();  
-      String getNoMasstraceInfoIsotopePattern();
-      int getIsotopePatternIterations();
-      int getCandidates();
-      int getTopNHits();
+      const vector<String> callSiriusQProcess(const String& tmp_ms_file,
+                                              const String& tmp_out_dir,
+                                              String& executable,
+                                              const String& out_csifingerid,
+                                              const bool decoy_generation);
+
+    private:
+    class ParameterModifier
+    {
+      const String openms_param_name;
+      SiriusAdapterAlgorithm *enclose;
+
+    public:
+      explicit ParameterModifier(const String &param_name, SiriusAdapterAlgorithm *enclose) :
+              openms_param_name(param_name), enclose(enclose) {}
+
+      void withValidStrings(initializer_list<String> choices)
+      {
+        enclose->defaults_.setValidStrings(openms_param_name, choices);
+      }
+
+      void withMinInt(int value)
+      {
+        enclose->defaults_.setMinInt(openms_param_name, value);
+      }
+    };
+
+    class ParameterSection
```

Comment:
This almost looks like a full general abstraction of command line tools and should probably go into a separate class.

Reply:
Maybe. SIRIUS since 4.4 is now actually not one algorithm/command-line tool anymore, but a growing aggregation of different algorithms, which can be called either in dependence on each other or independently. 

This abstraction allows the integration of additional algorithms/tools. Currently, SIRIUS, FingerID & passatutto are integrated. Later we could integrate ZODIAC as well. Of course at some point the SiriusAdapter might just blow up.

Reply:
I just mean, that by providing a list of subcommands and a mapping from TOPPBase parameters to tool parameters and with parameter modifiers, these classes could be generalized to be used for any command-line tool that we are wrapping. But I don't have a strong opinion on that. Can stay here for now.